internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -748,7 +761,16 @@ func setRootFields(d *schema.ResourceData, cluster *admin.ClusterDescription2024
 	if err := d.Set("global_cluster_self_managed_sharding", cluster.GetGlobalClusterSelfManagedSharding()); err != nil {
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "global_cluster_self_managed_sharding", clusterName, err))
 	}
-
+	if cluster.ConfigServerType != nil {
```

Comment:
just to confirm that we don't need to check for empty strings

Reply:
Not sure what case you are thinking of 😅
This is a computed field from backend and even if it is the empty string wouldn't we still want ot set it?

Reply:
makes sense